### ISSUE TITLE
[docs] Issue with 'HTTP cache'

### ISSUE THREAD

**Issue by y-a-v-a** [webhintio/hint#3072]
https://webhint.io/docs/user-guide/hints/hint-http-cache/index.html

I was receiving a hint with this description: 'The directive "no-store" is not recommended' but in the docs I cannot find anything about "no-store", this makes it hard to understand the "why" of this hint.

**Comment by Malvoz** [comment]
If there's insufficient documentation I agree it needs clarification, but just to try to answer your question about why `no-store` is "bad": You're asking the HTTP cache not to store anything about the response, this means more bandwidth is wasted on repeated requests for that resource, but also latency due to having to fetch the resource over the network instead of retrieving it from the cache.

Can I ask what you're trying to do with `no-store`? I think most people use the directive when they want to make sure that a resource is up-to-date, in that case you could use [`no-cache`](https://tools.ietf.org/html/rfc7234#section-5.2.2.2) instead (i.e. check with the server if there is a more recent version of the resource, if not then use the current one from the cache).

Here are some resources on Cache-Control that you may find helpful:

- https://www.mnot.net/cache_docs/
- https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching
- https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching
- https://csswizardry.com/2019/03/cache-control-for-civilians

**Comment by antross** [comment]
Yep, we should cover that in our docs, thanks @y-a-v-a for reporting!

And thanks @Malvoz for explaining :)

@molant Looks like you added this check as part of implementing https://github.com/webhintio/hint/issues/142. From reading the ["no-store" section of HTTP Caching](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching#no-cache_and_no-store) referenced from our documentation, it seems this directive could still be valid in cases where a request/response contains sensitive information that should not be committed to disk. Do you recall if detecting this as an exception was discussed? Apart from that I completely agree with this recommendation and updating our docs to explain it.

**Comment by molant** [comment]
> Do you recall if detecting this as an exception was discussed? 

There wasn't any discussion. The reasoning behind was that:

* this directive will not come up on first load of pages which is the most common scenario (also, how can we detect that something contains sensitive information?)
* if it is really required the user will know and then it can be deactivated via `ignoredUrls` and specifying the hint to disable

> Apart from that I completely agree with this recommendation and updating our docs to explain it.

Do we want to just explain this directive or should we have a small section for each one?

**Comment by antross** [comment]
> Do we want to just explain this directive or should we have a small section for each one?

I think we should have a small section for each one. Since there aren't that many reported by the hint, having the summary for each directly in our documentation will make this easier for users.

**Comment by y-a-v-a** [comment]
@Malvoz I think my company's devops team has added the `no-store` to HTML pages that contain specific state, so indeed we don't want it to be cached.

Thanks for the efforts to extend the documentation!
